ts/Dashboard/EditMode/Sidebar.ts

```diff
@@ -72,7 +73,52 @@ class Sidebar {
     public static components: Array<Sidebar.AddComponentDetails> = [
         {
             text: 'layout',
-            onDrop: function (): void {}
+            onDrop: function (
+                sidebar: Sidebar,
+                dropContext: Cell|Row
+            ): Cell|void {
+
+                if (!dropContext) {
+                    return;
+                }
+
+                const row = (
+                        dropContext.getType() === 'cell' ?
+                            (dropContext as Cell).row :
+                            (dropContext as Row)
+                    ),
+                    dashboard = row.layout.dashboard,
+                    newLayoutName = GUIElement.createElementId('layout'),
+                    cellName = GUIElement.createElementId('cell'),
+                    layout = new Layout(dashboard, {
+                        id: newLayoutName,
+                        copyId: '',
+                        parentContainerId: dashboard.container.id,
+                        rows: [{
+                            cells: [{
+                                id: cellName
+                            }]
+                        }],
+                        style: {}
+                    });
+
+                if (layout) {
+                    dashboard.layouts.push(layout);
+                }
+                const component = Bindings.addComponent({
```

Comment:
Wouldn't it be better to add an HTML component?

Reply:
I figured this is less code to add. (In the HTML component there is this long string 😓. I can change if you like)

Reply:
IMO HTML will look better but I'm going to leave it to @sebastianbochan